packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -1007,6 +1007,11 @@ class RenderParagraph extends RenderBox
       }
     }
 
+    assert(() {
+      _textPainter.debugPaintTextLayoutBoxes = debugPaintTextLayoutBoxes;
```

Comment:
Just curious, why doesn't `TextPainter.paint` just use the global `debugPaintTextLayoutBoxes` directly instead of an intermediary `TextPainter.debugPaintTextLayoutBoxes`? Feel free to keep as-is, asking for my own knowledge :)

Reply:
Hmm for no particular reason actually. I initially wanted to implement the logic in `RenderParagraph` so I defined the flag in `rendering/debug.dart` but then I changed my mind since adding the flag to `TextPainter` would allow `RenderEditable` to reuse the logic. 

Right now only static text (i.e. `RenderParagraph`s) reads the flag, not text in the text fields, since I'm not really sure if it will be useful for text fields.

Reply:
Thanks for your thoughts, I appreciate it! :)